src/coprocessor/codec/convert.rs

```diff
@@ -45,51 +86,157 @@ pub fn truncate_f64(mut f: f64, flen: u8, decimal: u8) -> Res<f64> {
 
 /// `overflow` returns an overflowed error.
 macro_rules! overflow {
-    ($val:ident, $bound:ident) => {{
-        Err(box_err!("constant {} overflows {}", $val, $bound))
+    ($val:ident, $bound:tt) => {{
+        Error::Eval(
+            format!("constant {} overflows {}", $val, $bound),
+            ERR_DATA_OUT_OF_RANGE,
+        )
     }};
 }
 
+/// `convert_int_to_int` converts an int value to a diferent int value.
+#[inline]
+pub fn convert_int_to_int(ctx: &mut EvalContext, val: i64, tp: FieldTypeTp) -> Result<i64> {
+    let lower_bound = integer_signed_lower_bound(tp);
+    // https://dev.mysql.com/doc/refman/8.0/en/out-of-range-and-overflow.html
+    if val < lower_bound {
+        ctx.handle_overflow(overflow!(val, lower_bound))?;
```

Comment:
in TiDB no warnings will be generated?

Reply:
See: https://github.com/pingcap/tidb/blob/627bd69235f68f8c388929bc103f5bf02c04bdbd/types/convert.go#L117

Reply:
Yes, in TiDB it even does not accept a context. How would there be a warning?

Reply:
The caller will handle this(even though not all cases) if the overflow error returned.
Some missing handling should be fixed in the TiDB.